bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -512,9 +515,11 @@ private BookKeeper(ClientConfiguration conf,
         // initialize ledger manager
         try {
             this.ledgerManagerFactory =
-                LedgerManagerFactory.newLedgerManagerFactory(conf, ((ZKRegistrationClient) regClient).getZk());
+                LedgerManagerFactory.newLedgerManagerFactory(conf, regClient.getLayoutManager());
         } catch (KeeperException ke) {
```

Comment:
if we remove zk, `LedgerManagerFactory.newLedgerManagerFactory` should not throw KeeperException.

Reply:
thanks. will change it.

Reply:
right, will remove this.